### DISCUSSION TITLE
Commandline: compare a file with a file in ZIP

### DISCUSSION THREAD

**Discussion by Pierre-de-la-Verre** [WinMerge/winmerge#1605] [General]
The current help file shows under Chapter "Commandline - unpack:" only TBD, so maybe you can help me here.

I need a commandline to compare
- "c:\data\test.txt" with
- test.txt in "d:\archive\compressed_files.zip"

Is it possible?

Thanks and regards!

**Comment by sdottaka** [comment]
The command line option /unpacker specifies the unpacker plugin and is not related to compressed files.

Unfortunately, WinMerge cannot directly compare a file and a file within a Zip file. 

However, it is possible to compare a folder and a folder within a Zip file, as shown below:

c:\progra~1\winmerge\winmergeu.exe /r c:\data d:\archive\compressed_files.zip /f test.txt

**Comment by Pierre-de-la-Verre** [comment]
Thanks.
This command does exactly what I asked for - it compares the file "test.txt", located in a folder and in a ZIP. Or am I on the wrong way?

**Reply by sdottaka** [reply]
Note that the above command will expand all the files in the zip file once and then compare them.
Also, if the test.txt file is contained in the folder `data` in d:\archive\compressed_files.zip, specify the following

`c:\progra~1\winmerge\winmergeu.exe /r c:\data d:\archive\compressed_files.zip\data /f test.txt`

**Comment by Pierre-de-la-Verre** [comment]
Thanks, OK for me.
